homeassistant/components/proxmox/sensor.py

```diff
@@ -0,0 +1,143 @@
+"""Support for sensors to show the resource usages of Proxmox VE."""
+import homeassistant.components.proxmox as proxmox
+from homeassistant.helpers.entity import Entity
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up Proxmox VE sensors."""
+    nodes = hass.data[proxmox.DATA_PROXMOX_NODES]
+    sensors = []
+
+    for name in nodes.keys():
+        item = nodes[name]
+        if 'type' in item and item['type'] == 'node':
+            sensors.append(
+                PXMXSensor(
+                    hass, name, 'Uptime', uptime, "min", 'mdi:clock-outline'))
+        else:
+            sensors.append(
+                PXMXSensor(hass, name, 'VCPU Count', vcpus, None, 'mdi:chip'))
+        sensors.append(
+            PXMXSensor(hass, name, 'Memory', ram, "%", 'mdi:memory'))
+        sensors.append(
+            PXMXSensor(hass, name, 'Disk', disk, "%", 'mdi:harddisk'))
+        sensors.append(
+            PXMXSensor(hass, name, 'CPU', cpu, "%", 'mdi:chip'))
+        sensors.append(
+            PXMXSensor(
+                hass, name, 'Max. Memory', mem_max, 'GB', 'mdi:memory'))
+        sensors.append(
+            PXMXSensor(
+                hass, name, 'Memory Used', mem_used, 'GB', 'mdi:memory'))
+        sensors.append(
+            PXMXSensor(
+                hass, name, 'Max. Disk', disk_max, 'GB', 'mdi:harddisk'))
+        sensors.append(
+            PXMXSensor(
+                hass, name, 'Disk Used', disk_used, 'GB', 'mdi:harddisk'))
+        sensors.append(
+            PXMXSensor(hass, name, 'Status', status, None, None))
+
```

Comment:
My suggestion would be to make one sensor per vm with the status and add all other stuff (memory, CPU, disk used etc.) as attributes of that sensor.
I have installed this component and get a huge load of sensors added, which makes things a bit overwhelming.

Other item for the attributes can be to show if the VM is a template.